assets/src/edit-story/app/media/utils/getResourceFromMedia3p.js

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Internal dependencies
+ */
+import createResource from './createResource';
+
+function getUrls(m) {
+  if (m.type.toLowerCase() === 'image') {
+    return Object.fromEntries(
+      new Map(
+        m.imageUrls.map((u) => [
+          u.imageName,
```

Comment:
what happens if this conflicts?

Reply:
Good question. Currently we're not returning the imageNames as well so it wouldn't work well. I'd like to tackle this in a separate PR as there's non-trivial logic to determine the "full"
, "large" and "web_stories_thumbnail" values. I've created #3065 to track this.